### PR TITLE
refactor: #1328 remove mapStateToProps and change to hooks

### PR DESCRIPTION
**Pull request by tanphamhaiduong**
Changes
- Remove mapStateToProps and change to react hooks
- Refactor for admin approvals state

# Pull request checklist

## Does this close any currently open issues?

fixes: #1328 

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #1328 

## What is the new behavior?

- Refactor for approval revision page
- Add more test

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import { REVISIONS_PER_PAGE } from '@/constants/paginator'
+import { appDetailRequestData } from '@/actions/app-detail'
+import { ApprovalModel, AppRevisionModel } from '@reapit/foundations-ts-definitions'
+import { AppDetailModel } from '@/types/marketplace-api-schema'
+import { selectApp } from '@/selector/app-detail'
+import { selectAppRevision } from '@/selector/app-revisions'
+import AdminApprovalModal from '@/components/ui/admin-approval-modal'
+import { selectAdminApprovalsState, selectWaitingApprovalData } from '@/selector/admin'
+
+export const handleCloseModal = ({ setIsModalOpen }) => () => setIsModalOpen(false)
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx:26]: Please add params type definition

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+export const handleOnPageChange = history => page => history.push(`${routes.ADMIN_APPROVALS}?page=${page}`)
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx:28]: Please add params type definition

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+export type HandleViewDetailOnClick = {
+  dispatch: Dispatch<any>
+  appId: string
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+  appRevisionId: string
+  currentRevisionId?: string
+  currentAppId?: string
+}
+
+export const handleViewDetailOnClick = ({
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx:80]: Please add params type definition

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  setIsModalOpen(true)
+}
+
+export type RenderViewDetailButton = {
+  revisionDetail: AppRevisionModel
+  appDetail: AppDetailModel
+  setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+  dispatch: Dispatch<any>
+}
+export const renderViewDetailButton = ({
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx:105]: Please add params type definition

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        setIsModalOpen,
+        appId,
+      })}
+    >
+      View details
+    </Button>
+  )
+}
+
+export const generateTableColumn = ({
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx:133]: Please add params type definition

**Comment by vuhuucuong** [comment]
Is there something wrong with this file `packages/marketplace/src/components/pages/admin-approvals/__tests__/__snapshots__/admin-approvals.test.tsx.snap`
Why does it have `+1,043,512` line. Can you check again?

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx (hunk 3 above)
Comment (tanphamhaiduong) [reply packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx]: This one also have type definition for params

File: packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx (hunk 5 above)
Comment (tanphamhaiduong) [reply packages/marketplace/src/components/pages/admin-approval/admin-approvals.tsx]: This one also have type definition

File: packages/marketplace/src/components/pages/admin-approvals/__tests__/admin-approvals.test.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        setIsModalOpen: jest.fn(),
+        dispatch: jest.fn(),
+      }
+      const result = generateTableColumn(mockProps)
+      expect(result).toHaveLength(5)
+    })
+  })
+
+  describe('AdminApprovals', () => {
+    const mockState = {
```
Comment (willmcvay) [inline packages/marketplace/src/components/pages/admin-approvals/__tests__/admin-approvals.test.tsx:120]: There is a stub for app state in the reducers folder - can we extend that please just so each test file doesn't have a copy-pasted state object in it. Will make tests more maintainable when we change the state object

File: packages/marketplace/src/selector/__tests__/admin.ts [hunk 7]
```diff
@@ -1,22 +1,71 @@
...
-    const input = {
-      adminApps: {
-        adminAppsData: appsDataStub.data,
-        loading: false,
-      },
-    } as ReduxState
-    const result = selectAdminAppsState(input)
-    expect(result).toEqual({ adminAppsData: appsDataStub.data, loading: false })
+describe('admin', () => {
+  const mockState = {
```
Comment (willmcvay) [inline packages/marketplace/src/selector/__tests__/admin.ts:12]: As above, can we have a central state stub to extend for out tests?

File: packages/marketplace/src/selector/__tests__/app-revision.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
+import { ReduxState } from '@/types/core'
+import { revisionsDataStub } from '@/sagas/__stubs__/revisions'
+import { selectAppRevision, selectAppRevisions, selectAppRevisionDetail } from '../app-revisions'
+
+describe('app-revision', () => {
+  const mockState = {
```
Comment (willmcvay) [inline packages/marketplace/src/selector/__tests__/app-revision.ts:6]: Same comment, would be nice to have a central state mock, then just override keys we need to change with `...rest`

**Comment by tanphamhaiduong** [comment]
@willmcvay @vuhuucuong @nphivu414 fixed as your comment

**Pull request merged**